Add templates feature: layout, list, item components, Template type, …

nshan20 opened the pull request:
…and /templates route

nshan20 committed:
Add templates feature: layout, list, item components, Template type, and /templates route

nshan20 committed:
Merge branch 'angular-19' into ag19-264

aamialkovich commented on indigo-frontend/src/app/app.routes.ts:
```diff
@@ -55,7 +56,7 @@ export const routes: Routes = [
         ],
       },
       {
-        path: 'projects/:projectId/notebooks/:notebookId',
```
Revert please

aamialkovich commented on indigo-frontend/src/app/app.routes.ts:
```diff
@@ -87,47 +88,12 @@ export const routes: Routes = [
         data: { requiredRole: 'Dictionary editor' },
       },
       {
-        path: 'projects/:projectId/notebooks/:notebookId/experiments/:experimentId',
```
Also revert this and add your route

aamialkovich commented on indigo-frontend/src/core/types/entities/template.i.ts:
```diff
@@ -1,63 +1,29 @@
+import { Attachment } from './attachment.i';
+import { ProjectAcl } from './acl.i';
 
-export interface TemplateComponentAttachments {
```
Revert this

aamialkovich commented on indigo-frontend/src/core/types/entities/template.i.ts:
```diff
@@ -1,63 +1,29 @@
+  inherited?: boolean;
 }
 
 export interface Template extends BaseEntity {
```
Seems not correct. Please refer to https://indigo-eln-dev.test.lifescience.opensource.epam.com/swagger/eln/#/Template%20Resource/get_api_eln_templates

aamialkovich commented on indigo-frontend/src/core/types/entities/template.i.ts:
```diff
@@ -1,63 +1,29 @@
-  | TemplateComponentIntendedProducts;
-
-export interface TemplateTab {
+export interface TeamMember {
```
Remove this

aamialkovich commented on indigo-frontend/src/app/pages/template/template-list/template-list.component.ts:
```diff
@@ -0,0 +1,109 @@
+import { Component, inject, Input, OnDestroy } from '@angular/core';
```
Please get rid of project-related stuff like opening project creation dialog and so on.

aamialkovich commented on indigo-frontend/src/core/components/template/template-item/template-item.component.ts:
```diff
@@ -0,0 +1,47 @@
+  ngOnInit(){
+    console.log(this.template);
+  }
+  mock_users = [
```
don't need many users here as there is only one creator

aamialkovich commented on indigo-frontend/src/core/components/template/template-item/template-item.component.ts:
```diff
@@ -0,0 +1,47 @@
+    '-',
+  ];
+  @Input() template: Template;
+  @Input() variant: 'grid' | 'list' = 'grid';
```
Seems no need two variants as there is only the one on mockups,

nshan20 committed:
Remove TeamMember interface as requested in PR review

nshan20 committed:
Fix: apply requested changes from review

nshan20 committed:
WIP: staged changes before rebase

aamialkovich commented on indigo-frontend/src/app/app.routes.ts:
```diff
@@ -59,74 +60,81 @@ export const routes: Routes = [
           },
         ],
       },
+      // {
```
It shouldn't be commented.

aamialkovich commented on indigo-frontend/src/core/components/util/infinite-scroll.base.ts:
```diff
@@ -12,6 +12,7 @@ export abstract class InfiniteScrollBase<T> extends PaginatedBase<T> {
     this.config.enableScrollRestoration = true;
     this.data$ = this.dataList$.pipe(
       switchMap((data) => {
+        console.log(data, "000000000000000000000");
```
Remove console.log

aamialkovich commented on indigo-frontend/src/app/pages/template/template-list/template-list.component.html:
```diff
@@ -0,0 +1,52 @@
+  />
+</div>
+
+<eln-button
```
Do we need it here?

aamialkovich commented on indigo-frontend/src/app/pages/template/template-list/template-list.component.ts:
```diff
@@ -0,0 +1,128 @@
+  }
+
+  async openModal() {
+    const ref = this.dialog.open(ProjectAddComponent);
```
We don't need projectadd dialog here

nshan20 committed:
Add TemplateService and delete API; wire template list and item delete; remove debug log; restore experiment route

nshan20 committed:
Render template list from ItemTemplate array; update InfiniteScrollBase generic; tidy delete callback

- Change template loop to iterate over data directly (data instead of data['items'])
- Import and use ItemTemplate as InfiniteScrollBase generic instead of RootTemplate
- Remove stray blank line inside deleteTemplate subscribe callback

aamialkovich commented on package-lock.json:
Remove please

aamialkovich commented on indigo-frontend/src/core/types/entities/template.i.ts:
```diff
@@ -61,3 +73,34 @@ export interface Template extends BaseEntity {
+  displayName: string
+}
+
+export interface ModifiedBy {
```
Seems that the same as CreatedBy, so can be extend from it.

aamialkovich commented on indigo-frontend/src/core/services/api.service.ts:
```diff
@@ -47,6 +47,11 @@ export class ApiService<T> {
     return this.httpClient.get<T>(this.buildUrl(dictionary));
   }
 
+  public delete(url: string, id: string ): Observable<void> {
```
Seems that void should be replaces with T

aamialkovich commented on indigo-frontend/src/core/services/template/template.service.ts:
```diff
@@ -0,0 +1,32 @@
+    return this.api.getPaged('templates', pager, { search });
+  }
+
+  createTemplate(body: any) {
```
Please check in swagger type and get rid of any

aamialkovich commented on indigo-frontend/src/core/services/template/template.service.ts:
```diff
@@ -0,0 +1,32 @@
+  }
+
+  deleteTemplate(templateId: string) {
+    return this.api.delete(`templates/`, templateId,);
```
last comma is unnecessary

aamialkovich commented on indigo-frontend/src/core/components/template/template-item/template-item.component.ts:
```diff
@@ -0,0 +1,38 @@
+  variant = 'list';
+
+  deleteTemplate(templateId: string) {
+    this.templateService.deleteTemplate(templateId).subscribe(() => {
```
```suggestion
    this.templateService.deleteTemplate(templateId).subscribe()
```

aamialkovich commented on indigo-frontend/src/core/components/template/template-item/template-item.component.ts:
```diff
@@ -0,0 +1,38 @@
+export class TemplateItemComponent {
+  constructor(private templateService: TemplateService) {}
+
+  mock_users = {
```
can we delete it?

nshan20 committed:
Add CreateTemplate types and type createTemplate body; replace ModifiedBy with CreatedBy; make ApiService.delete generic; simplify template deletion call/subscription

aamialkovich committed:
refactoring
